From: Andrew Morton <akpm@linux-foundation.org>
Subject: kasan-add-shadow-checks-to-wrappers-and-rename-kasan_arch_is_ready-fix
Date: Sat Jul 19 05:01:18 PM PDT 2025

fix build

>> mm/kasan/tags.c:145:6: error: redefinition of 'kasan_save_free_info'
     145 | void kasan_save_free_info(struct kmem_cache *cache, void *object)
         |      ^
   mm/kasan/kasan.h:403:20: note: previous definition is here
     403 | static inline void kasan_save_free_info(struct kmem_cache *cache, void *object)
         |                    ^

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202507190528.W6ndwfyg-lkp@intel.com/
Cc: Alexander Gordeev <agordeev@linux.ibm.com>
Cc: Alexander Potapenko <glider@google.com>
Cc: Andrey Konovalov <andreyknvl@gmail.com>
Cc: Andrey Ryabinin <ryabinin.a.a@gmail.com>
Cc: Christophe Leroy <christophe.leroy@csgroup.eu>
Cc: Dmitriy Vyukov <dvyukov@google.com>
Cc: Heiko Carstens <hca@linux.ibm.com>
Cc: Sabyrzhan Tasbolatov <snovitoll@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/kasan/tags.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/kasan/tags.c~kasan-add-shadow-checks-to-wrappers-and-rename-kasan_arch_is_ready-fix
+++ a/mm/kasan/tags.c
@@ -142,7 +142,7 @@ void kasan_save_alloc_info(struct kmem_c
 	save_stack_info(cache, object, flags, false);
 }
 
-void kasan_save_free_info(struct kmem_cache *cache, void *object)
+void __kasan_save_free_info(struct kmem_cache *cache, void *object)
 {
 	save_stack_info(cache, object, 0, true);
 }
_
